### PR TITLE
feat(tools): publish 15 member-family fields and declare 9 GitLab never sends

### PR DESCRIPTION
**Pull request by jmrplens**
## Description

The sent dimension of the 1:1 review reached the Member family: access requests, group members, billable members and invitations. Fifteen of the fields it named are published from the captured response under [ADR-0021](docs/development/adr/adr-0021-captured-response-for-fields-the-sdk-does-not-model.md). The other nine turn out not to be sent at all, and are answered rather than surfaced.

`AccessRequestExtra` embeds `MemberExtra` rather than restating it, because `API::Entities::AccessRequester` really does inherit `Member` and merge `UserBasic`, so eight keys arrive through the embed. The object-valued fields reuse the shapes the sibling member packages already publish on the same json keys, so no third copy of an identity or a member role enters the audit.

`invite_token` is in the JSON and deliberately absent from the Markdown. A rendered table would paste a live credential into the conversation, which is the reasoning `conan_package_name` already set on the other side. Whether the JSON should drop it too is under discussion on a review thread.

## Related Issue

No single issue: this is one tranche of the continuing 1:1 field review, whose upstream half is tracked in <https://gitlab.com/gitlab-org/api/client-go/-/issues/2300>.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [x] Enhancement (improvement to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring (no functional change)
- [ ] Documentation update
- [ ] CI / build / tooling

## Changes Made

- Publishes 15 fields across `accessrequests`, `groupmembers`, `groups` and `invites`, each read from the captured response because client-go models none of them.
- Declares nine fields as not sent, under `entity-option-no-endpoint-passes`, a category that existed and had never been used. `avatar_path`, `custom_attributes` and `is_using_seat` are gated on presenter options that no route these types serve declares. A presenter option is not a request parameter a caller can smuggle in: Grape passes it only where the endpoint declares it, which is what separates these from a field gated on a permission, where the same endpoint does send it to somebody. The member lists are the counterpart and prove the split is per route set rather than per entity: `GET /groups/:id/members` and `/members/all` do declare `show_seat_info`, so `is_using_seat` stays published there.
- Declares nine more against `GET /groups/:id/billable_members`, whose Grape `desc` says `Entities::Member` while the handler presents `BillableMember`. They are named one by one rather than with a wildcard, because `groupmembers` publishes the real `Member` elsewhere and a wildcard would swallow real findings.
- Adds a guard test per affected type: a fixture that spells the option-gated keys and an assertion that none reaches the surface, so the code cannot drift back from the declaration.
- Names the three member-family package paths as constants in the declaration table, which Sonar reads as duplicated literals and a reader has to compare by eye.
- Clears 85 mutants that survived in `internal/tools/groups` before this change, in push rules, hooks, sharing, avatar and provisioned users.
- Regenerates the README counts, the llms listings, the token footprint and the testing reference.

## How to Test

1. `go build ./...` and `go test ./internal/... -count=1`.
2. `go run ./cmd/audit_1to1/ -scope=paths 2>&1 | ...` and read `summary.typed_unsurfaced_fields`: it falls from 1388 to 1373, exactly the fifteen published, with `stale_declarations` 0.
3. Filter `shapes.typed.unsurfaced` to the four packages: every remaining row carries a `category`, none is undeclared.
4. `make coverage-mutants PKG=./internal/tools/accessrequests` and the same for `groupmembers`, `invites` and `groups`, with `--workers 2 --timeout-coefficient 20`.

## Breaking Changes / Migration Notes

N/A. Every change is additive to a response, or a declaration in a tool that reports rather than gates.

### Code Quality

- [x] Code compiles: `go build ./...`
- [x] Consolidated Go analysis passes on changed packages
- [x] Code is formatted
- [x] Follows idiomatic Go patterns and the conventions documented in `CLAUDE.md` / `.github/instructions/`

### Testing

- [x] All existing tests pass
- [x] New tests added for new functionality (table-driven, with `httptest` mocks)
- [x] Coverage on modified packages is at the project's bar: every touched package reports `Lived 0`, `Not covered 0` and `Timed out 0` under gremlins, and full condition coverage under gobco
- [x] Edge cases and error scenarios covered, including both sides of every condition and the captured-read failure
- [ ] If applicable, E2E tests updated/added under `test/e2e/suite/` — not applicable, no action or route changes

### Documentation

- [x] Doc comments added for exported types/functions (godoc)
- [x] `docs/` updated where the change is recorded
- [x] `README.md` regenerated
- [ ] If introducing/removing a tool — none added or removed
- [x] Commit messages follow Conventional Commits

### Security

- [x] No secrets, tokens, or credentials in code, tests, fixtures, or logs. One field of the API's own response is an invitation token, published in JSON and kept out of the Markdown; it is under discussion on a review thread.
- [x] Input validation for user-provided parameters
- [x] Error messages do not leak sensitive information
- [x] No new dependencies

### CODE REVIEW

File: cmd/audit_1to1/internal/paths/sent_declarations.go [hunk 1]
```diff
@@ -69,6 +121,36 @@ var declaredUnsurfaced = []sentDeclaration{ //nolint:gochecknoglobals // the adj
...
+	{Package: toolsDir + "/groupmembers", Entity: memberEntity, Field: "avatar_path", Category: categoryOptionNeverPassed, Reason: reasonOnlyPathNeverPassed},
+	{Package: toolsDir + "/groupmembers", Entity: memberEntity, Field: "custom_attributes", Category: categoryOptionNeverPassed, Reason: reasonCustomAttributesNeverPassed},
+	{Package: toolsDir + "/groups", Entity: memberEntity, Field: "avatar_path", Category: categoryOptionNeverPassed, Reason: reasonOnlyPathNeverPassed},
+	{Package: toolsDir + "/groups", Entity: memberEntity, Field: "custom_attributes", Category: categoryOptionNeverPassed, Reason: reasonCustomAttributesNeverPassed},
+
+	// is_using_seat is declared for the access requests alone. The group and
+	// project member lists do declare show_seat_info, so the same key on those
+	// types is published from the SDK rather than answered here.
+	{Package: toolsDir + "/accessrequests", Entity: memberEntity, Field: "is_using_seat", Category: categoryOptionNeverPassed, Reason: reasonShowSeatInfoNeverPassed},
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply cmd/audit_1to1/internal/paths/sent_declarations.go]: Checked against the report, and the entity is right as it stands: changing it to `accessRequesterEntity` would be what breaks the match.

The record annotates each finding with the entity the field was **read on**, not with the entity the route presents. Running the audit on this branch:

```
field=avatar_path        entity=API::Entities::AccessRequester  category=entity-option-no-endpoint-passes
field=custom_attributes  entity=API::Entities::AccessRequester  category=entity-option-no-endpoint-passes
field=is_using_seat      entity=API::Entities::Member           category=entity-option-no-endpoint-passes
```

`avatar_path` and `custom_attributes` come from the `UserBasic` that `AccessRequester` merges, so they are read on the requester. `is_using_seat` comes from `ee/lib/ee/api/entities/member.rb`, which extends `Member`, so it is read on `Member` even though the route presents an `AccessRequester`. All three are classified, and the summary confirms it: the in-scope filter returns 0 undeclared findings.

That is why the table uses two different entity constants for one package, and the comment above the entry says so.

File: llms-full-meta-tools.txt [hunk 2]
```diff
@@ -419,15 +419,15 @@ Annotations: readOnly=false, destructive=true, idempotent=false, openWorld=true
...
+{"additionalProperties":false,"properties":{"invitations":{"items":{"additionalProperties":false,"properties":{"access_level":{"type":"integer"},"created_at":{"type":"string"},"created_by_name":{"type":"string"},"expires_at":{"type":"string"},"id":{"type":"integer"},"invite_email":{"type":"string"},"invite_token":{"type":"string"},"user_name":{"type":"string"}},"required":["id","invite_email","access_level"],"type":"object"},"type":["null","array"]},"next_steps":{"items":{"type":"string"},"type":["null","array"]},"pagination":{"additionalProperties":false,"properties":{"has_more":{"type":"boolean"},"next_page":{"type":"integer"},"page":{"type":"integer"},"per_page":{"type":"integer"},"prev_page":{"type":"integer"},"total_items":{"type":"integer"},"total_pages":{"type":"integer"}},"required":["page","per_page","total_items","total_pages","next_page","prev_page","has_more"],"type":"object"}},"required":["invitations","pagination"],"type":"object"}
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply llms-full-meta-tools.txt:422]: Thank you for flagging this one; it is the finding on this pull request worth the most discussion, so I would rather not resolve it quickly in either direction.

The context that is not visible from the diff: this server's stated purpose is a 1:1 surface over the GitLab API, and the field is exactly what `GET /groups/:id/invitations` sends. `lib/api/entities/invitation.rb` exposes `invite_token` unconditionally, so any caller holding a token that can list invitations already receives it from GitLab directly. Dropping it here does not remove their access to it; it removes it only from this path, and this project treats a field GitLab sends that we do not publish as a defect.

Against that, your point stands on its own terms: a token that can be redeemed is not the same class of value as a name, and it currently reaches a structured client that may log or persist a whole tool result without anyone deciding to.

There is already a partial concession in the change: the token is deliberately kept out of the Markdown rendering, so it is not pasted into a conversation transcript, and the code says why. Whether that is the right stopping point, or whether the JSON should drop it too and accept a documented deviation from the 1:1 rule, is the maintainer's call rather than mine. I have put it to him and will act on the answer here.

Leaving this thread open until then rather than closing it.

[the remark this answers is not in the document]
Comment (jmrplens) [reply llms-full-meta-tools.txt:422]: The maintainer has decided: `invite_token` stays in the JSON.

The reasoning, now recorded beside the field in `internal/tools/invites/invites.go` rather than only here, so the next reviewer finds it where they look: GitLab sends this token to any caller whose credential can already list invitations, so removing it denies nobody the value and only makes this server misreport what the endpoint returns, which this project treats as a defect of its own. The Markdown exclusion is where the line is drawn, because a transcript is the surface a person reads without having asked for it.

Your framing of the risk was the useful part and is preserved in that comment: the concern is a structured client persisting the value without anyone deciding to, which is a different thing from the caller's own access.

Thank you for pressing on it. Resolving.
